calico/felix/test/test_fetcd_api.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014, 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+felix.test.test_etcd_api
+~~~~~~~~~~~
+
+Top level tests for EtcdAPI.
+"""
+import logging
+import socket
+import gevent
+from etcd import EtcdException
+from mock import Mock, call, patch
+from calico.datamodel_v1 import key_for_status, key_for_uptime
+from calico.felix.fetcd import EtcdAPI
+from calico.felix.test.base import BaseTestCase
+
+_log = logging.getLogger(__name__)
+
+# To make testing continuous status reporting more convenient,
+# we speed up sleeping.
+_oldsleep = gevent.sleep
+def _newsleep(duration):
+    _oldsleep(duration * 0.01)
+gevent.sleep = _newsleep
+
+
+class TestEtcdAPI(BaseTestCase):
+    def setUp(self):
+        super(TestEtcdAPI, self).setUp()
+        self.m_config = Mock()
+        self.m_config.IFACE_PREFIX = "tap"
+        self.m_config.ETCD_ADDR = "localhost:4001"
+        self.m_config.HOSTNAME = socket.gethostname()
+        self.m_config.RESYNC_INTERVAL = 0
+        self.m_config.REPORTING_INTERVAL_SECS = 0
+        self.m_config.REPORTING_TTL_SECS = 0
+
+    @patch('calico.felix.fetcd.EtcdAPI.write_to_etcd')
+    @patch('calico.felix.fetcd._EtcdWatcher')
+    @patch('calico.felix.fetcd.etcd')
+    def finish_setup(self, m_etcd, m_EtcdWatcher, m_write_to_etcd, **kwargs):
+        # Set configuration attributes and start etcd_api
+        for key, value in kwargs.iteritems():
+            setattr(self.m_config, key, value)

```

Comment:
This isn't wrong, but I think it could be neater. Rather than updating the key/value pairs one-by-one, you could just use the update() method:

> <code><strong>update</strong>([<em>other</em>])</code>

> Update the dictionary with the key/value pairs from other, overwriting existing keys. Return None. ([Python docs](https://docs.python.org/2/library/stdtypes.html#dict.update))

So instead you'd write:

```python
self.m_config.update(kwargs)
```